Sources/Auth/Services/Wallet/AuthRequestSubscriber.swift

```diff
@@ -0,0 +1,37 @@
+import Combine
+import Foundation
+import WalletConnectUtils
+
+class AuthRequestSubscriber {
+    private let networkingInteractor: NetworkInteracting
+    private let logger: ConsoleLogging
+    private var publishers = [AnyCancellable]()
+    private let messageFormatter: SIWEMessageFormatting
+    var onRequest: ((_ id: Int64, _ message: String)->())?
+
+    init(networkingInteractor: NetworkInteracting,
+         logger: ConsoleLogging,
+         messageFormatter: SIWEMessageFormatting) {
+        self.networkingInteractor = networkingInteractor
+        self.logger = logger
+        self.messageFormatter = messageFormatter
+        subscribeForRequest()
+    }
+
+    private func subscribeForRequest() {
+        networkingInteractor.requestPublisher.sink { [unowned self] subscriptionPayload in
+            guard subscriptionPayload.request.method == "wc_authRequest" else { return }
+            guard let authRequest = try? subscriptionPayload.request.params?.get(AuthRequestParams.self) else {
+                logger.debug("Malformed auth request params")
+                return
+            }
+            do {
+                let message = try messageFormatter.formatMessage(from: authRequest)
+                guard let requestId = subscriptionPayload.request.id?.right else { return }
```

Comment:
@MisterVants I began to integrate you rpc lib and I do not like this part of querying id from request. Do you guys think we need `Either` type for request id? It is totally unexpected to have a String type request id in the network.

@flypaper0 ⬆️

Reply:
I doesn't know do we really need Either or not, but we can make it CustomStringConvertible to work with `id` like with regular String

Reply:
But may be better to make id - String, but make possibility to parse Int/String id param from RPC response

Reply:
You should not query the request ID internal value as an Int or String. The `RPCID` type already abstracts that and should be treated as a primitive. Just use it as you would use an integer ID:

`var onRequest: ((_ id: RPCID, _ message: String)->())?`

`guard let requestId = subscriptionPayload.request.id else { ... }`

Reply:
(Might need to `import JSONRPC`)

Reply:
I am not sure if it is a good idea to make all classes passing RPCID up to the app and make them all dependent to `JSONRPC`. 
Another inconvenience is that id is an optional

Reply:
The typealias you suggested might be a good idea to avoid imports, and helps decoupling in case of any future changes. It's a primitive, after all.

The optional ID is required by the jsonrpc 2.0 specs to expect notification requests. Unfortunately, this is essential to make the RPC layers future-proof